src/Microsoft.TestPlatform.Common/Filtering/FilterExpression.cs

```diff
@@ -296,19 +294,23 @@ internal static FilterExpression Parse(string filterString)
                 throw new FormatException(string.Format(CultureInfo.CurrentCulture, CommonResources.TestCaseFilterFormatException, CommonResources.MissingOperator));
             }
 
+            if (fastFilterBuilder.ContainsValidFilter)
```

Comment:
ToFastFilter already seems to have this check.

Reply:
You are right.